packages/browser/src/transports/base.ts

```diff
@@ -68,12 +68,17 @@ export abstract class BaseTransport implements Transport {
   /**
    * @inheritDoc
    */
-  public recordLostEvent(type: Outcome, category: SentryRequestType): void {
+  public recordLostEvent(reason: Outcome, category: SentryRequestType): void {
     if (!this.options.sendClientReport) {
       return;
     }
-    const key = `${type}:${CATEGORY_MAPPING[category]}`;
-    logger.log(`Adding ${key} outcome`);
+    // We want to track each category (event, transaction, session) separately
+    // but still keep the distinction between different type of outcomes.
+    // We could use nested maps, but it's much easier to read and type this way.
+    // A correct type for map-based implementation if we want to go that route
+    // would be `Partial<Record<SentryRequestType, Partial<Record<Outcome, number>>>>`
```

Comment:
I still miss _why_ `[category]:[reason]` is the format: the header reserves `:` as the separator, so as long as `category` and `reason` don't have `:` is fine.

Reply:
It can be `[category]|[reason]`, `[category]/[reason]` or anything else. It's just for the storage purpose. We split it for the delivery anyway on line 116

Reply:
Understood incorrectly 👍.